test/Elastic.Apm.AspNetCore.Tests/AspNetCoreDiagnosticListenerTest.cs

```diff
@@ -194,5 +194,45 @@ public async Task CaptureDiagnosticAndHostingUnhandledExceptionEvent()
 				context?.Request.Method.Should().Be(HttpMethod.Get.Method);
 			}
 		}
+
+
+		[Fact]
+		public async Task RecordFullRequestPathWhenFullPathRequestMatchingIsConfigured()
+		{
+			var capturedPayload = new MockPayloadSender();
+			using (var agent = new ApmAgent(new TestAgentComponents(payloadSender: capturedPayload, configuration: new MockConfiguration(useFullPathRequestMatching: "true"))))
+			{
+				// Based on Helper GetClient
+				var builder = _factory
+					.WithWebHostBuilder(n =>
+					{
+						n.Configure(app =>
+						{
+							app.Use((context, next) =>
+							{
+								context.Request.PathBase = "/ReverseProxyPrefix";
+								return next();
+							});
```

Comment:
This will not test `AspNetCoreDiagnosticListener` - it'll only test the `ApmMiddleware`.

`AspNetCoreDiagnosticListener` is a diagnostic source based implementation of ASP.NET Core tracing without an ASP.NET Core middleware. The main reason we have it is because that's the only way to enable the agent with zero code change (we can't register a middleware - or it'd be fairly hard).

The test should cover both.